libs/authorizenet/src/drivers/authorize-net.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable, Inject } from '@angular/core';
+import { Store } from '@ngrx/store';
+
+	import { 
+		DaffPaymentServiceInterface,
+		DaffPaymentTransformer,
+		DaffPaymentTransformerService
+	} from '@daffodil/checkout';
+
+import { DaffAuthorizeNetGenerateTokenSuccess, DaffAuthorizeNetGenerateTokenFailure } from '../actions/authorizenet.actions';
+import { AuthorizeNetResponse } from '../models/response/authorize-net-response';
+import { DaffAuthorizeNetTokenRequest } from '../models/request/authorize-net-token-request';
+
+declare var Accept: any;
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DaffAuthorizeNetPaymentService implements DaffPaymentServiceInterface<DaffAuthorizeNetTokenRequest> {
+
+	constructor(
+		@Inject(DaffPaymentTransformer) public paymentTransformer: DaffPaymentTransformerService<DaffAuthorizeNetTokenRequest>,
+		private store: Store<any>
+	) {}
+
+	generateToken(paymentTokenRequest: DaffAuthorizeNetTokenRequest): string {
+		Accept.dispatchData(this.paymentTransformer.transformOut(paymentTokenRequest), this.responseHandler.bind(this));
```

Comment:
This should be an observable, there's a way to port this `node.js` style callback to an observable:

https://stackoverflow.com/questions/43131162/making-an-observable-from-a-callback

Reply:
done